plugin/interface_generator/main.go

```diff
@@ -141,6 +141,33 @@ func FieldListDestruct(structPrefix string, fieldList *ast.FieldList, fileset *t
 	return strings.Join(result, ", ")
 }
 
+func FieldListDestructErrors(structPrefix string, fieldList *ast.FieldList, fileset *token.FileSet) string {
+	if fieldList == nil || len(fieldList.List) == 0 {
+		return ""
+	}
+
+	result := ""
+	nextLetter := 'A'
+	for _, field := range fieldList.List {
+		typeNameBuffer := &bytes.Buffer{}
+		err := printer.Fprint(typeNameBuffer, fileset, field.Type)
+		if err != nil {
+			panic(err)
+		}
+		typeName := typeNameBuffer.String()
+		if strings.Contains(typeName, "error") || strings.Contains(typeName, "Error") {
+			result = structPrefix + string(nextLetter)
+			break
+		}
+		nextLetter++
+	}
+
+	if result == "" {
+		return "true"
+	}
+	return fmt.Sprintf("%s == nil", result)
```

Comment:
It will clash when `true` is returned.

Reply:
The main reason I have done it this way is because some functions return ` *model.AppError` or `error` type where what you are saying is works to check if the error is `nil`. 

But in some places there is `*http.Response` or `bool` or some other _non-error_ return value. There directly `true` should be returned even when there are return fields. Hence, I have done it this way. 

And finally, there are some functions which don't return any values and `true` should be returned. This is handled by the `... {{ else }} ...` part.

What do you think?

Reply:
Ah I see. Thanks for the explanation. Could we just merge case 1 and case 3? and just return `true` for both from this function. That will help eliminate the `{{ else }}` from the calling part and keep all the logic here. What do you think? @Ashniu123